### PR TITLE
grab sdl2 include automatically

### PR DESCRIPTION
**Pull request by therealssj**
Fixes: automatically grab sdl2 include path

Note: on macOS sdl2-config returns:
`-I/usr/local/include/SDL2 -D_THREAD_SAFE`
while on ubuntu it returns:
`-I/usr/local/include/SDL2 -D_REENTRANT`

what is the difference between the two?

Does this change need to mentioned in CHANGELOG.md?
no

Requires changes in protobuff specs?
no

Requires testing
no

### CODE REVIEW

File: tiny-firmware/Makefile [hunk 1]
```diff
@@ -117,7 +117,7 @@ CFLAGS   += -DHEADLESS=1
 else
 CFLAGS   += -DHEADLESS=0
 
-CFLAGS   += -I$(SDL_INCLUDE) -D_REENTRANT
```
Comment (olemis) [inline tiny-firmware/Makefile]: [needed by firmware](https://stackoverflow.com/questions/2601753/what-is-the-reentrant-flag) , pls revert

File: tiny-firmware/Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 APPVER = 1.0.0
 
-SDL_INCLUDE ?= /usr/include/SDL2
+SDL_INCLUDE := $(shell sdl2-config --cflags)
```
Comment (olemis) [inline tiny-firmware/Makefile]: Please use `?=` to allow for overriding this value if provided .

File: tiny-firmware/Makefile [hunk 3]
```diff
@@ -117,7 +117,7 @@ CFLAGS   += -DHEADLESS=1
 else
 CFLAGS   += -DHEADLESS=0
 
-CFLAGS   += -I$(SDL_INCLUDE) -D_REENTRANT
+CFLAGS   += $(SDL_INCLUDE) -D_REENTRANT
```
Comment (olemis) [inline tiny-firmware/Makefile]: why `-I` removed ?

File: tiny-firmware/emulator/Makefile [hunk 4]
```diff
@@ -17,7 +18,8 @@ ifeq ($(HEADLESS),1)
 CFLAGS   += -DHEADLESS=1
 else
 CFLAGS   += -DHEADLESS=0
-CFLAGS   += -I$(SDL_INCLUDE) -I$(TOOLCHAIN_DIR)/include -D_REENTRANT
+CFLAGS   += $(SDL_INCLUDE) -D_REENTRANT
```
Comment (olemis) [inline tiny-firmware/emulator/Makefile]: why `-I` removed ?

File: tiny-firmware/Makefile (hunk 3 above)
Comment (therealssj) [reply tiny-firmware/Makefile]: please see the note in pr description. the output of sdl2-config already has `-I`

**Comment by therealssj** [comment]
is it fine if both `-D_THREAD_SAFE` and `-D_REENTRANT` flags are define @olemis ?

**Comment by olemis** [comment]
> is it fine if both `-D_THREAD_SAFE` and `-D_REENTRANT` flags are defined @olemis ?

the firmware only requires `-D_REENTRANT`

**Comment by therealssj** [comment]
@olemis fixed 🎉

**Comment by olemis** [comment]
@therealssj I already merged (most of) your changes with `develop`. There are merge conflicts here , but most of the changes are already solved (`SDL_INCLUDE` => `SDL_CFLAGS`). Can we close this PR?

**Pull request closed without merging**